plugin/trino-kafka-event-listener/src/test/java/io/trino/plugin/eventlistener/kafka/TestKafkaEventListenerPlugin.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.eventlistener.kafka;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.spi.eventlistener.EventListenerFactory;
+import io.trino.testing.kafka.TestingKafka;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+
+public class TestKafkaEventListenerPlugin
+{
+    private static final String CREATED_TOPIC = "query_created";
+    private static final String COMPLETED_TOPIC = "query_completed";
+    private static TestingKafka testingKafka;
+
+    @BeforeAll
+    public static void setUp()
+    {
+        testingKafka = TestingKafka.create();
+        testingKafka.start();
+        testingKafka.createTopic(CREATED_TOPIC);
+        testingKafka.createTopic(COMPLETED_TOPIC);
+    }
+
+    @AfterAll
+    public static void teardown()
+            throws IOException
+    {
+        testingKafka.close();
+    }
+
+    @Test
+    void testCreateEventListener()
+    {
+        KafkaEventListenerPlugin plugin = new KafkaEventListenerPlugin();
+
+        EventListenerFactory factory = getOnlyElement(plugin.getEventListenerFactories());
+        factory.create(
+                ImmutableMap.<String, String>builder()
+                        .put("kafka-event-listener.publish-created-event", "true")
+                        .put("kafka-event-listener.publish-completed-event", "true")
+                        .put("kafka-event-listener.broker-endpoints", testingKafka.getConnectString())
+                        .put("kafka-event-listener.created-event.topic", CREATED_TOPIC)
+                        .put("kafka-event-listener.completed-event.topic", COMPLETED_TOPIC)
+                        .put("kafka-event-listener.env-var-prefix", "INSIGHTS_")
+                        .put("kafka-event-listener.request-timeout", "30s")
+                        .put("kafka.security-protocol", "plaintext")
+                        .buildOrThrow())
```

Comment:
Please also do e2e test where `EventListenerFactory` sends an event and it's consumed via `TestingKafka`? This might require adding some `TestingKafka#receive` method

Reply:
Good idea - Added an e2e test to write and read back a created/completed event